src/types.ts

```diff
@@ -229,3 +229,17 @@ export interface LocalizeMixin {
   hass?: HomeAssistant;
   localize: LocalizeFunc;
 }
+
+interface ForecastAttribute {
+  temperature: number;
+  templow?: number;
+  precipitation?: number;
+  humidity?: number;
+}
+
+export type WeatherEntity = HassEntityBase & {
+  attributes: HassEntityAttributeBase & {
+    temperature: number;
+    forecast?: ForecastAttribute;
```

Comment:
I have made this an array in my local code. I have unfinished code for hourly forecast, awaiting bram reply before pushing. Just an FYI I know this is wrong hehe